runtime/parser/type.go

```diff
@@ -467,6 +478,34 @@ func defineIntersectionOrDictionaryType() {
 			}
 		},
 	)
+
+	// While restricted types have been removed from Cadence, during the first few months of the
+	// migration period, leave a special error in place to help developers
+	// TODO: remove this after Stable Cadence migration period is finished
```

Comment:
While at it, maybe also create an issue to revert this, so we don't forget later